code/Model/RedirectorPage.php

```diff
@@ -171,35 +175,12 @@ public function syncLinkTracking()
         }
     }
 
-    protected function onBeforeWrite()
```

Comment:
Need to keep this is as you can programatically save things without it going via a UrlField

Reply:
That felt very much like it was being done _instead_ of validation. I feel like validating the value in the better option.

I can add the URL validation in the `validate()` method for this class if you like? But I don't think it makes sense to alter the value that's being set, especially programatic values.

Reply:
OK makes, yes agree throwinga validation exception is better than setting the value to blank and writing